Added secret token to combat impersonation.

LukeTheVirus opened the pull request:
- Added secret token used by the client to authorise api calls. Prevents others from doing api calls with the public avaliable login id. Secret gets created and passed to client when creating or joining a lobby.

- Removed /rooms route, not sure if this should be public.

- Added various type hinting.

Client side pull request: https://github.com/Geekid812/TrackmaniaBingo/pull/17

Commit:
Added client_secret to prevent impersonation.

- Added secret token used by the client to authorise api calls. Prevents impersonation with only login token provided by ubisoft. Secret gets created when host creates a lobby or someone joins a lobby.

- Removed /rooms route, not sure if this should be public.

- Added various type hinting.

Geekid812 commented on routes/claim.py:
```diff
@@ -7,6 +7,7 @@ async def claim_cell(request: web.Request):
     body = loads(await request.text())
     room, player = GameServer.instance().find_player(body['login'])
     if room is None: return web.Response(status=404)
+    if player.secret != body['client_secret']: return web.Response(status=405)
```
The HTTP code 405 suggests "Method not allowed" which I'm not sure is the most suitable here. That doesn't really matter though as we're not exposing this in a public API.

Geekid812 commented on routes/create.py:
```diff
@@ -19,5 +20,8 @@ async def create(request: web.Request):
+    
+    return web.Response(text=dumps({
+        'room_code': room.code, 
+        'client_secret': host.secret
```
I think it might be worth doing the secret exchange in the TCP opening handshake instead of over the HTTP API. That's where identification is handled, and supposedly authentification should be there too.

Geekid812 commented on routes/start.py:
```diff
@@ -1,11 +1,14 @@
 
 async def start(request: web.Request):
-    login = await request.text()
+    body = loads(await request.text())
```
In the opening handshake, the client would send their login and receive a unique token. In subsequent requests, there would then be no need to send the login again, only the token is required.

Geekid812 reviewed (commented):
I appreciate the initiative here, but I think soon I'll replace the HTTP API with two-sided TCP communication with a WebSocket-like protocol which will make this authentification problem nonexistant. There's a lot of good changes here though, so I will probably merge this so that it serves as a base for the refactoring.

Commit:
Switched from login to full token authorisation.
- Fixed hosts being able to start game even when maps are not loaded.

LukeTheVirus commented:
Thanks for the feedback. 
I implemented some of your proposed changes, mainly switching over to only using the random token.
I also implemented a fix for the start button so people don't get confused when the server hasn't loaded the maps from tmx yet.

Geekid812 reviewed (approved):
Seems good! :+1:
